File: Linguagens de Programação/GOLang/prime.go
package main
import ("fmt")

func main() {
    fmt.Println("Calculadora de número primo")
	fmt.Println("Digite o número de Base: ")
    var n int 
    fmt.Scanln(&n)
	var result bool = true
	for i := 2; i < n; i++ {
		if n % i == 0 {
			result = false
			break
		}
	}
	if result {
		fmt.Println("O número é primo")
	} else {
		fmt.Println("O número não é primo")
	}
}

File: Linguagens de Programação/Haskell/prime.hs
isPrime :: Int -> Bool
isPrime n = n > 1 && all (\d -> n `mod` d /= 0) [2..n-1]

main = do
    putStrLn "Digite um número inteiro positivo:"
    n <- readLn
    if isPrime n
        then putStrLn (show n ++ " é primo.")
        else putStrLn (show n ++ " não é primo.")
